source/security/authentication.txt

```diff
@@ -0,0 +1,554 @@
+certificates as well as TLS/SSL options, see our
+:ref:`TLS/SSL guide <kotlin-sync-tls>`.
+
+.. _kotlin-sync-auth-aws:
+
+MONGODB-AWS
+-----------
+
+.. note::
+
+   The MONGODB-AWS authentication mechanism is available for MongoDB
+   deployments on MongoDB Atlas.
+
+The ``MONGODB-AWS`` authentication mechanism uses your Amazon Web Services
+Identity and Access Management (AWS IAM) credentials to authenticate your
+user. To learn more about configuring MongoDB Atlas, see the
+:atlas:`Set Up Authentication with AWS IAM </security/aws-iam-authentication>`
+guide.
+
+To instruct the driver to use this authentication mechanism, you can specify
+``MONGODB-AWS`` either as a parameter in the connection string or by using
+the ``MongoCredential.createAwsCredential()`` factory method.
+
+Learn how to specify this authentication mechanism and the various ways to
+provide your AWS IAM credentials in the next sections.
+
+These sections contain code examples that use the following placeholders:
+
+* ``awsKeyId`` - value of your AWS access key ID
+* ``awsSecretKey`` - value of your AWS secret access key
+* ``atlasUri`` - network address of your MongoDB Atlas deployment
+* ``hostname`` - hostname of your MongoDB Atlas deployment
+* ``port`` - port of your MongoDB Atlas deployment
+* ``awsSessionToken`` - value of your AWS session token
+
+.. _kotlin-mongodb-aws-sdk:
+
+AWS SDK 
+~~~~~~~
+
+You can use one of the AWS SDK for Java v1 or v2 to specify your credentials.
+This method offers the following features:
+
+- Multiple options for obtaining credentials
+- Credential caching which helps your application avoid rate limiting
+- Credential provider management for use with the `Elastic Kubernetes Service <https://docs.aws.amazon.com/eks/latest/userguide/what-is-eks.html>`__.
+
+To use the AWS SDK for Java for ``MONGODB-AWS`` authentication, you must
+perform the following:
+
+1. Specify the authentication mechanism
+#. Add the SDK as a dependency to your project
+#. Supply your credentials by using one of the methods in the credential
+   provider chain
+
+To specify the authentication mechanism by using a ``MongoCredential``,
+use the ``MongoCredential.createAwsCredential()`` factory method
+and add the ``MongoCredential`` instance to your ``MongoClient`` as shown
+in the following example:
+
+.. literalinclude:: /includes/security/authentication.kt
+   :language: kotlin
+   :dedent:
+   :start-after:  start-aws-sdk-mcred
+   :end-before: end-aws-sdk-mcred
+   :emphasize-lines: 1, 9
+
+To specify the authentication mechanism in the connection string, add
+it as a parameter as shown in the following example:
+
+.. literalinclude:: /includes/security/authentication.kt
+   :language: kotlin
+   :dedent:
+   :start-after:  start-aws-sdk-cred-string
+   :end-before: end-aws-sdk-cred-string
+
+To add the AWS SDK as a dependency to your project, see the following
+AWS documentation for the version you need:
```

Comment:
Q: Do you know when a user would need v1 vs. v2?

Reply:
Just depending on which java version the user wants to use. Looked into it more, and the java v1 version is reaching end of support next year. Included a note at the top of section to recommend using v2 instead.